node/pkg/driver/node_utils.go

```diff
@@ -580,6 +597,27 @@ func (d NodeUtils) GetBlockVolumeStats(mpathDevice string) (VolumeStatistics, er
 	return volumeStats, nil
 }
 
+func (d NodeUtils) IsVolumePathMatchesVolumeId(volumeId string, volumePath string) (bool, error) {
+	volumeUuid := d.GetVolumeUuid(volumeId)
+	volumeIdByVolumePath, err := d.osDeviceConnectivityHelper.GetVolumeIdByVolumePath(volumePath, volumeUuid)
+	if err != nil {
+		return false, err
+	}
+	logger.Infof("IsVolumePathMatchesVolumeId: found volume id [%s] for volume path [%s] ", volumeId, volumePath)
+
+	return d.isVolumeIdFromPathIsMatched(volumeIdByVolumePath, volumeUuid), nil
+}
+
+func (d NodeUtils) isVolumeIdFromPathIsMatched(volumeIdByVolumePath string, volumeUuid string) bool {
+	volumIds := d.osDeviceConnectivityHelper.GetVolumeIdVariations(volumeUuid)
+	for _, volumeId := range volumIds {
+		if strings.Contains(volumeIdByVolumePath, volumeId) {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
this look like a duplicate of `isSameId`

Reply:
`isSameId` checks exactly the strings,  `isVolumeIdFromPathIsMatched ` checks if it's contains it

Reply:
will continue this conversation [here](https://github.com/IBM/ibm-block-csi-driver/pull/491#discussion_r882430317)